File: 2025-2/Arquitetura de Computadores/np-01/questao-01.c
#include <stdio.h>

int main()
{
    //Declara array e inicia um loop que vai iterar 20 vezes 
    int evenNumbers[20];

    for (int i = 0; i < 20;)
    {
        //Imprime texto na tela e pede pro usuario digitar um numero
        printf("Digite um numero par \n");
        scanf("%d", &evenNumbers[i]);
        //Verifica se o numero NAO eh par
        if (evenNumbers[i] % 2 != 0)
        {
            // Imprime texto na tela e usa o continue para saltar para a proxima iteracao
            printf("Numero digitado nao eh par. \n");
            continue;
        }
        // Adicionar 1 na variavel do loop. 
        i++;
    }
    //Declara variavel
    int avg;
    // Loop com 20 iteracoes
    for (int i = 0; i < 20; i++)
    {
        //soma o valor digitado pelo usuario a variavel avg
        avg += evenNumbers[i];
    }
    // divide a variavel por 20 para obter a media
    avg = avg / 20;
    // Imprime o valor da media no console.
    printf("A media dos numeros digitados eh: %d", avg);
    return 0;
}

File: 2025-2/Laboratório de Programação de Computadores/N01/questao-01.py
# importa biblioteca random do python
import random
# declara array vazio
numbers = []
# loop para iterar 15 vezes
for _ in range(15):
    # pendura inteiro random entre 1 e 200 no fim do array 
    numbers.append(random.randint(1,200))

# Declara variavel para armazenar a soma dos numeros    
sum = 0

# Loop para iterar entre os elementos do array
for number in numbers:
    # condicional para saber se o numero eh multiplo de 4 e maior que 50
    if number % 4 == 0 and number > 50:
        # soma valor do numero a variavel soma
        sum += number
        # imprime o numero na tela
        print(number)
        
# imprime o valor da variavel soma        
print("soma:", sum)